core/instance/namespace.go

```diff
@@ -40,12 +35,11 @@ type namespace struct {
 // model UUID. NewNamespace returns an error if the model tag is invalid.
 func NewNamespace(modelUUID string) (Namespace, error) {
 	if !names.IsValidModel(modelUUID) {
-		return nil, errors.Errorf("model ID %q is not a valid model", modelUUID)
+		return nil, errors.NotValidf("model UUID %q", modelUUID)
 	}
-	// The suffix is the last six hex digits of the model uuid.
-	suffix := modelUUID[len(modelUUID)-uuidSuffixDigits:]
```

Comment:
I am not sure if it is worth changing to use a prefix for instance namespace.
The instance namespace is used for machine tags/instance names/hostname etc.
We will have to consider implementing some sort of upgrade steps to avoid breakage.

Reply:
revert to use suffix for namespace.name.